### PR TITLE
Reservation Screen UI Design.

### PR DESCRIPTION
**Pull request by ankur141295**
@RivuChk This pull request, once merged, will add the Reservation Success UI to the app. 
Here are a few details to note:

-  The flight recommendation image card has a white corner radius because the Figma image used had the same issue.
-  The invoice card's custom shape is not yet implemented. This is a work in progress for which I'll create another pull request.

**Figma UI**
Screenshot 2024-04-20 at 8 15 01 AM

**Designed UI**
Screenshot_20240420_081833

### CODE REVIEW

File: app/src/main/java/me/darthwithap/hotel_app/presentation/reservation/success/ReservationSuccessScreen.kt [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+        )
+
+        Spacer(modifier = Modifier.height(8.dp))
+
+        Text(
+            modifier = Modifier.padding(horizontal = 52.dp),
+            textAlign = TextAlign.Center,
+            text = buildAnnotatedString {
+                withStyle(
+                    style = AppTheme.typography.forms16Regular.toSpanStyle()
```
Comment (p42rthicle) [inline app/src/main/java/me/darthwithap/hotel_app/presentation/reservation/success/ReservationSuccessScreen.kt]: Here, in the figma file, title14 typography is used, please check

**Review by p42rthicle — COMMENTED** [review]
Good work, I was wondering if we should introduce state variables instead of hard coded strings.
Setting up dummy viewModel layer etc.?

File: app/src/main/java/me/darthwithap/hotel_app/presentation/reservation/success/ReservationSuccessScreen.kt (hunk 1 above)
Comment (ankur141295) [reply app/src/main/java/me/darthwithap/hotel_app/presentation/reservation/success/ReservationSuccessScreen.kt]: @patthhar I have changed the font size according to Figma. Please have a look again.

**Comment by ankur141295** [comment]
> Good work, I was wondering if we should introduce state variables instead of hard coded strings. Setting up dummy viewModel layer etc.?

We certainly can! However, I'm currently awaiting information about the Reservation Success Response Model before creating the Model class.

Additionally, I wanted to consult with @RivuChk regarding the appropriate scope for a ViewModel when using Voyager. I recently came across this article [https://medium.com/androiddevelopers/dependency-injection-in-compose-a2db897e6f11] and was curious about how it applies to our situation.

Screenshot 2024-04-20 at 10 31 43 AM

**Pull request merged**